libs/utils/analysis/stune_analysis.py

```diff
@@ -0,0 +1,129 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2015, ARM Limited and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+""" SchedTune Analysis Module """
+
+import matplotlib.gridspec as gridspec
+import matplotlib.pyplot as plt
+import numpy as np
+import pylab as pl
+
+from analysis_module import AnalysisModule
+from devlib.utils.misc import memoized
+from trappy.utils import listify
+
+# Configure logging
+import logging
+
+
+class STuneAnalysis(AnalysisModule):
+    """
+    Support for SchedTune signals anaysis
+
+    :param trace: input Trace object
+    :type trace: :mod:`libs.utils.Trace`
+    """
+
+    def __init__(self, trace):
+        super(STuneAnalysis, self).__init__(trace)
+
+###############################################################################
+# DataFrame Getter Methods
+###############################################################################
+
+    @memoized
+    def _dfg_tasks_updates(self, cpus=None, cgroups=None):
+        if not self._trace.hasEvents('sched_tune_tasks_update'):
+            logging.warn('Events [sched_tune_tasks_updates] not found, '
+                         'cannot get schedtune tasks updates!')
+            return None
+
+        # Get SchedTune Accouting data
+        df = self._dfg_trace_event('sched_tune_tasks_update')
+
+        # Get number of CPUs to plot
+        if cpus:
+            # Filter on requested CPUs
+            cpus = listify(cpus)
+            df = df[df.cpu.isin(cpus)]
+
+        # Get number of CGROUPs to plot
+        if cgroups:
+            # Filter on requested CGROUPs
+            cgroups = listify(cgroups)
+            df = df[df.idx.isin(cgroups)]
+
+        return df
+
+
+###############################################################################
+# Plotting Methods
+###############################################################################
+
+    def plotTasksUpdate(self, cpus=None, cgroups=None, xlim=(0,None)):
+        """ Plot SchedTune accouting on specified cpus/cgroups
```

Comment:
I think so far I've only seen these formats of docstrings in LISA:
```
""" blabla """
```
and
```
"""
blabla
"""
```

I think yours is still PEP8-friendly, but for some reason it triggers my OCD :(